packages/grid/x-data-grid/src/internals/components/panel/filterPanel/GridFilterPanel.tsx

```diff
@@ -93,9 +93,13 @@ function GridFilterPanel(props: GridFilterPanelProps) {
 
   const deleteFilter = React.useCallback(
     (item: GridFilterItem) => {
+      const shouldCloseFilterPanel = items.length === 1;
       apiRef.current.deleteFilterItem(item);
+      if (shouldCloseFilterPanel) {
+        apiRef.current.hideFilterPanel();
```

Comment:
If the developer is controlling the filter items through the `filterModel` prop we don't know if he accept the new value passed via `onFilterModelChange`. In this case we can't assume that when `hideFilterPanel` is called it will have 0 filter items. I had the same problem with the `editRowsModel` prop triggering synchronous actions after changing the prop value.

Reply:
I forget it was controllable. We could track `items.length` and close the pannel when reaching 0